### PR TITLE
Support UI cluster creation

### LINKED ISSUE #303: Support cluster creation from UI
We would like to support cluster creation from UI and not just CLI. Currently CLI will install the serviceaccount, role, role-binding to the managed cluster, but in order to support this from UI, we need to move this logic server side.

The ClusterCreateRequest will need a new field which takes a kubeconfig as a string. This kubeconfig is used temporarily to just to install the argocd-manager account, and thrown away at the end of the request.

### PR DESCRIPTION
**Pull request by merenbach**
Closes #303.

### CODE REVIEW

**Comment by merenbach** [comment]
@jessesuen requesting a structural review to make sure I have the right idea here: move everything from `install.go` into `util/db` and have all that done server-side.  The only thing I'm trying to figure out is what to do with the provided kubeconfig.

File: cmd/argocd/commands/cluster.go [hunk 1]
```diff
@@ -40,6 +39,24 @@ func NewClusterCommand(clientOpts *argocdclient.ClientOptions, pathOpts *clientc
 	return command
 }
 
+// DefaultKubeConfigDir returns the local configuration path for settings such as cached authentication tokens.
+func DefaultKubeConfigDir() (string, error) {
```
Comment (jessesuen) [inline cmd/argocd/commands/cluster.go]: We shouldn't invent our own method for resolving a kubeconfig. This method will not honor the --kubeconfig flag. The previous logic to get a kubeconfig contents using `config, err := configAccess.GetStartingConfig()` needs to be preserved.

Comment (merenbach) [reply cmd/argocd/commands/cluster.go]: @jessesuen have now restored that code.

File: server/cluster/cluster.go [hunk 2]
```diff
@@ -76,6 +82,53 @@ func (s *Server) Create(ctx context.Context, q *ClusterCreateRequest) (*appv1.Cl
...
+		clusterServer = common.KubernetesInternalAPIServerAddr
+	} else if cluster, ok := kubeconfig.Clusters[q.Context]; ok {
+		clusterServer = cluster.Server
+		clusterInsecure = cluster.InsecureSkipTLSVerify
+	} else {
+		return nil, status.Errorf(codes.Internal, "Context %s does not exist in kubeconfig", q.Context)
+	}
+
+	// Temporarily install RBAC resources for managing the cluster
+	defer func() {
```
Comment (jessesuen) [inline server/cluster/cluster.go]: We should not be removing these resources as part of adding a cluster. We need these resources to manage the cluster.

**Review by jessesuen — CHANGES_REQUESTED** [review]

File: server/cluster/cluster.go (hunk 2 above)
Comment (merenbach) [reply server/cluster/cluster.go]: Fixed, thank you!

File: server/cluster/cluster.go [hunk 3]
```diff
@@ -16,15 +19,17 @@ import (
...
-	db  db.ArgoDB
-	enf *rbac.Enforcer
+	kubeclientset kubernetes.Interface
+	db            db.ArgoDB
+	enf           *rbac.Enforcer
 }
 
 // NewServer returns a new instance of the Cluster service
-func NewServer(db db.ArgoDB, enf *rbac.Enforcer) *Server {
+func NewServer(kubeclientset kubernetes.Interface, db db.ArgoDB, enf *rbac.Enforcer) *Server {
```
Comment (jessesuen) [inline server/cluster/cluster.go]: Per comment below, cluster server should not need a kubeclientset.  We create a clientset on the fly from a kubeconfig.

**Review by jessesuen — CHANGES_REQUESTED** [review]

**Review by jessesuen — APPROVED** [review]

**Pull request merged**